### ISSUE TITLE
Add subagents to the evo plugin (`plugins/evo/agents/`)

### ISSUE THREAD

**Issue by alokwhitewolf** [evo-hq/evo#50]
The evo plugin currently exposes 8 skills (`discover`, `optimize`, `finetuning`, `ideator`, `subagent`, `verifier`, `report`, `infra-setup`) that load text into the orchestrator's context. Several of these are better suited as proper subagents -- read-only roles with bounded scope, single-artifact output, and isolated context, invoked from the orchestrator via the host's Task tool.

The first plugin agent, `benchmark-reviewer`, shipped on `feat/model-update` (commit `ccd9217`); use it as a shape reference.

### What this adds

A `plugins/evo/agents/` directory alongside `plugins/evo/skills/`. Each agent is a `<name>.md` file with YAML frontmatter (`name`, `description`, `tools`, `model`) and a body that serves as the subagent's system prompt. Claude Code's plugin loader picks up `agents/<name>.md` files automatically; no manifest changes in `.claude-plugin/plugin.json` are required.

Invocation pattern from any skill:

```
Task(subagent_type=\"evo:<agent-name>\", prompt=\"...\")
```

### Skills worth converting

| Current skill | Why convert | Subagent shape |
|---|---|---|
| `verifier` | Read-only false-progress audit; returns pass/fail + findings. | Pattern A: invoked once, returns one JSON artifact. Orchestrator gates commits on the report. |
| `ideator` | Web-research-heavy (arXiv, HF Hub, GitHub, blogs). High token cost; pollutes orchestrator's context with research notes that aren't directly actionable. | Pattern A: invoked at decision points. Returns ranked proposals. |
| `subagent` (the spawn protocol) | Already IS the experiment subagent's system prompt; orchestrator constructs the brief inline today. | Pattern B: become the actual subagent definition. Orchestrator spawns via `Task(subagent_type=\"evo:experiment-runner\", prompt=brief)` instead of inline construction. |

`discover`, `optimize`, `finetuning`, `report`, `infra-setup` stay as skills -- they're orchestrator-level workflow guidance that has to be in-context for the orchestrator to plan against.

### Tradeoffs

- **Loss of in-context reference.** When `verifier` is a skill, the orchestrator carries the full checklist and can quote it back during reasoning. When it's a subagent, the orchestrator just knows \"invoke verifier, gate on its output.\" Mostly a win for skim-resistance, but the orchestrator can't introspect the verifier's logic without re-running it.
- **Two loader paths.** `agents/` and `skills/` are loaded by Claude Code separately; the plugin loader needs both functional. Confirmed working in `ccd9217`.
- **Skill deprecation.** Once moved, leave a stub `skills/<name>/SKILL.md` that redirects to the agent, so backward references in existing workspaces don't break.

### Migration order

1. `verifier` -> subagent (smallest; same shape as the existing `benchmark-reviewer`).
2. `ideator` -> subagent (biggest token-cost win; isolates web-fetch results from orchestrator context).
3. `subagent` skill -> `experiment-runner` subagent (biggest refactor; touches the `optimize` skill that uses it).

### Other agents this opens up

- `annotation-reviewer` -- audits the orchestrator's `evo annotate` entries for self-deceptive claims and missing pattern-detection (e.g. 2+ committed experiments at score 0.0 on a verifiable benchmark without a pivot annotation).
- `gate-reviewer` -- audits gate definitions for decorative-vs-real coverage (a benchmark-rerun gate that exits 0 just because the benchmark printed a score is decorative; a score-threshold gate that exits 1 below threshold is real).
- `dataset-leakage-checker` -- standalone version of the leakage section of `benchmark-reviewer`, callable on any training data source independently of the full benchmark audit.

**Comment by alokwhitewolf** [comment]
Shipped:
- benchmark-reviewer subagent added in ccd9217
- ideator subagent migrated in d69f35d (stub skill since removed in 8a8a778)
- verifier subagent migrated in 0127244 (stub skill since removed in 8a8a778)

All three live at `plugins/evo/agents/{benchmark-reviewer,ideator,verifier}.md` and are dispatched via Task tool with `subagent_type="evo:<name>"`. Available in 0.5.0-alpha.6.
